### ISSUE TITLE
Cannot use dependencies with special characters in their names

### ISSUE THREAD

**Issue by nguyer** [remoteinterview/zero#10]
Hi there,

I am trying to include a React component library called `@bandwidth/shared-components` It's on npmjs here: https://www.npmjs.com/package/@bandwidth/shared-components but when I include it in a .jsx file I get the following stack trace while `zero` tries to resolve dependencies:

```
✷ Updating packages(node:5410) UnhandledPromiseRejectionWarning: Error: Invalid package name "@bandwidth": name can only contain URL-friendly characters
    at invalidPackageName (/usr/local/lib/node_modules/npm/node_modules/npm-package-arg/npa.js:82:15)
    at Result.setName (/usr/local/lib/node_modules/npm/node_modules/npm-package-arg/npa.js:118:11)
    at new Result (/usr/local/lib/node_modules/npm/node_modules/npm-package-arg/npa.js:107:23)
    at Function.resolve (/usr/local/lib/node_modules/npm/node_modules/npm-package-arg/npa.js:54:15)
    at Object.keys.map (/usr/local/lib/node_modules/npm/lib/install/deps.js:466:20)
    at Array.map (<anonymous>)
    at BB.resolve.then (/usr/local/lib/node_modules/npm/lib/install/deps.js:465:47)
    at tryCatcher (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:512:31)
```

I'm guessing this is a wider-spread issue affecting more than just this library. I'm not sure if it's the `@` or the `/` that's breaking it right now.

I've poked around the `zero` source a little bit and could't find the error message that gets printed so I'm guessing it's coming from a library? What library would that be that is doing dependency management (or detection)?

**Comment by xkawi** [comment]
able to replicate this issue and found the root cause, but not sure how to fix it.

It's caused by the `npmi` dependency that's doing the dependency management (e.g. installing), and it also depends on `global-npm` which uses [npm-package-arg](https://github.com/npm/npm-package-arg/blob/a5a18b3752b0d107dc6e04527324d6f9aa289c04/npa.js#L114) to validate the package name.
The validation logic is located at this package called [validate-npm-package-name](https://github.com/npm/validate-npm-package-name/blob/80313b82a5c7e466556c3a7fb1cb7d7dab92f4be/index.js#L76) and somehow the package name validation fails.

Hope it helps to find a fix.

**Comment by xkawi** [comment]
Alright, found the line of code that causes it: https://github.com/remoteinterview/zero/blob/6677573d6b75f6a20d8747381e0cd9237c5a907a/packages/dep-tree-js/index.js#L50 that will trim the package name by “/“ character and only take the first index.

**Comment by nguyer** [comment]
Yep that would do it. Maybe we could try to validate the whole package name before doing the split and push that to `deps` if it's valid. Else, we could do what it currently does. I'll try that and if it works, open a PR.

**Comment by nguyer** [comment]
After playing around with the validation library, it is actually just using a regex. I think a simple check to see if the package name begins with `@` and not splitting if it does, will fix this. I opened https://github.com/remoteinterview/zero/pull/14 with that change. I haven't been able to actually verify it yet, because I'm still trying to figure out how to install zero from source. `npm install -g` from the root of the project directory doesn't seem to put it on my path. Any tips on that?

**Comment by asadm** [comment]
@nguyer I think running that inside packages/core would work as the `zero` bin is defined in that package.

This issue is fixed in v1.0.8 (06aa286f904a6292774d001539f0b4375caa437e)

**Comment by asadm** [comment]
@nguyer I have added npm script `npm run bootstrap` to automatically make `zero` command available system-wide. See [CONTRIBUTING.md](https://github.com/remoteinterview/zero/blob/master/CONTRIBUTING.md).
